WebDriverAgentLib/Categories/XCUIElement+FBTyping.m

```diff
@@ -141,24 +146,36 @@ - (BOOL)fb_clearTextWithSnapshot:(FBXCElementSnapshotWrapper *)snapshot
   NSString *placeholderValue = snapshot.placeholderValue;
   NSUInteger preClearTextLength = [currentValue fb_visualLength];
   do {
-    if (retry >= MAX_CLEAR_RETRIES - 1) {
-      // Last chance retry. Tripple-tap the field to select its content
-
-      if ([self respondsToSelector:@selector(tapWithNumberOfTaps:numberOfTouches:)]) {
-        // e.g. tvOS 17 raised unrecognized selector error for XCUIElementTypeSearchField
-        // while following typeText worked.
-        [self tapWithNumberOfTaps:3 numberOfTouches:1];
-      }
-      return [FBKeyboard typeText:backspaceDeleteSequence error:error];
-    }
-
-    NSString *textToType = [backspaceDeleteSequence fb_repeatTimes:preClearTextLength];
+    NSString *backspacesToType = [backspaceDeleteSequence fb_repeatTimes:preClearTextLength];
+#if TARGET_OS_IOS
+    // the ios needs to have keyboard focus to clear text
     if (shouldPrepareForInput && 0 == retry) {
       [self fb_prepareForTextInputWithSnapshot:snapshot];
     }
-    if (![FBKeyboard typeText:textToType error:error]) {
+
+    if (retry == 0) {
+      // 1st attempt is via the IOHIDEvent as a fastest operation
+      // https://github.com/appium/appium/issues/19389
+      [[XCUIDevice sharedDevice] fb_performIOHIDEventWithPage:0x07  // kHIDPage_KeyboardOrKeypad
+                                                        usage:0x9c  // kHIDUsage_KeyboardClear
+                                                     duration:0.01
+                                                        error:nil];
+    } else if (retry >= MAX_CLEAR_RETRIES - 1) {
+      // Last chance retry. Tripple-tap the field to select its content
+      [self tapWithNumberOfTaps:3 numberOfTouches:1];
+      return [FBKeyboard typeText:backspaceDeleteSequence error:error];
+    } else if (![FBKeyboard typeText:backspacesToType error:error]) {
+      // 2nd operation
+      return NO;
+    }
+#else
+    // tvOS does not need a focus
+    if (retry >= MAX_CLEAR_RETRIES - 1) {
+      return [FBKeyboard typeText:backspaceDeleteSequence error:error];
```

Comment:
does this make sense in tvOS? the whole text must be selected if we only want to tap backspace once. Otherwise we must tap it as many times as the count of chars in the input.

Reply:
Let me test a bit more with a couple of apps. Potentially for tvOS need only `typeText` one time

Reply:
I assume `[FBKeyboard typeText:backspacesToType error:error];` only is good for tvOS. I have tested common search field and email/password input view in tvOS. Not sure other type exists, so for now, we can do with this I believe